File: ObjC_Example/Fullboard/FullBoardAdMenuViewController.m
//
//  FullBoardAdMenuViewController.m
//  ObjC_Example
//
//  Copyright © 2017年 FAN Communications. All rights reserved.
//

#import "FullBoardAdMenuViewController.h"

static NSString *const CellIdentifier = @"Cell";

@interface FullBoardAdMenuViewController ()

@property (nonatomic) NSArray<NSString *> *items;
@property (nonatomic) NSArray<NSString *> *segues;
@property (nonatomic) NSArray<NSString *> *details;

@end

@implementation FullBoardAdMenuViewController

- (void)viewDidLoad
{
    [super viewDidLoad];
    
    self.title = @"FullBoard";
    
    self.items = @[@"インタースティシャル形式", @"スワイプ形式", @"スクロールエンド形式", @"タブ形式"];
    self.segues = @[@"PushDefault", @"PushPage", @"PushScrollEnd", @"PushTab"];
    self.details = @[@"ポップアップで表示された広告は右上の×ボタンにより閉じることができます。",
                     @"マンガや小説などスワイプでページ送りをするアプリにてページとページの間に広告を差し込むことができます。※×ボタンは非表示にできます。",
                     @"ニュースや記事まとめ、縦スクロール式のマンガアプリなどで最下部までスクロールした後に画面下部から広告を呼び出します。右上の×ボタンにて閉じることができます。",
                     @"ニュースや記事まとめアプリでカテゴリタブの中に\"PR\"タブを作成し、PRタブがタップされた際に広告を表示します。"];
}

#pragma mark - Table view data source

- (NSInteger)numberOfSectionsInTableView:(UITableView *)tableView
{
    // Return the number of sections.
    return 1;
}

- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section
{
    // Return the number of rows in the section.
    return self.items.count;
}

- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath
{
    UITableViewCell *cell = [tableView dequeueReusableCellWithIdentifier:CellIdentifier forIndexPath:indexPath];
    
    // Configure the cell...
    if (@available(iOS 14.0, *)) {
        UIListContentConfiguration *content = [cell defaultContentConfiguration];
        [content setText:self.items[indexPath.row]];
        [content setSecondaryText:self.details[indexPath.row]];
        [cell setContentConfiguration:content];
    } else {
        cell.textLabel.text = self.items[indexPath.row];
        cell.detailTextLabel.numberOfLines = 0;
        cell.detailTextLabel.text = self.details[indexPath.row];
    }
    cell.accessoryType = UITableViewCellAccessoryDisclosureIndicator;
    
    return cell;
}

#pragma mark - Table view delegate

- (void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath
{
    [self performSegueWithIdentifier:self.segues[indexPath.row] sender:nil];
}

#pragma mark - Navigation

- (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender
{
    UIViewController *viewController = segue.destinationViewController;
    viewController.view.autoresizingMask = UIViewAutoresizingFlexibleHeight;
}

@end

File: Swift_Example/Fullboard/FullBoardAdMenuViewController.swift
//
//  FullBoardAdMenuViewController.swift
//  ObjC_Example
//
//  Copyright © 2017年 FAN Communications. All rights reserved.
//

import UIKit

class FullBoardAdMenuViewController: UITableViewController {
    
    private let items = [
        (name: "インタースティシャル形式", segue: "PushDefault", detail: "ポップアップで表示された広告は右上の×ボタンにより閉じることができます。"),
        (name: "スワイプ形式", segue: "PushPage", detail: "マンガや小説などスワイプでページ送りをするアプリにてページとページの間に広告を差し込むことができます。※×ボタンは非表示にできます。"),
        (name: "スクロールエンド形式", segue: "PushWeb", detail: "ニュースや記事まとめ、縦スクロール式のマンガアプリなどで最下部までスクロールした後に画面下部から広告を呼び出します。右上の×ボタンにて閉じることができます。"),
        (name: "タブ形式", segue: "PushTab", detail: "ニュースや記事まとめアプリでカテゴリタブの中に\"PR\"タブを作成し、PRタブがタップされた際に広告を表示します。")
    ]
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        // Uncomment the following line to preserve selection between presentations
        // self.clearsSelectionOnViewWillAppear = false
        
        // Uncomment the following line to display an Edit button in the navigation bar for this view controller.
        // self.navigationItem.rightBarButtonItem = self.editButtonItem()
        
        self.title = "FullBoard"
    }
    
    // MARK: - Table view data source
    
    override func numberOfSections(in tableView: UITableView) -> Int {
        // Return the number of sections.
        return 1
    }
    
    override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        // Return the number of rows in the section.
        return self.items.count
    }
    
    override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
        let cell = tableView.dequeueReusableCell(withIdentifier: "Cell", for: indexPath)
        
        // Configure the cell...
        if #available(iOS 14.0, *) {
            var content = cell.defaultContentConfiguration()
            content.text = self.items[indexPath.row].name
            content.secondaryText = self.items[indexPath.row].detail
            cell.contentConfiguration = content
        } else {
            cell.textLabel!.text = self.items[indexPath.row].name
            if let detailTextLabel = cell.detailTextLabel {
                detailTextLabel.numberOfLines = 0
                detailTextLabel.text = self.items[indexPath.row].detail
            }
        }
        cell.accessoryType = .disclosureIndicator
        
        return cell
    }
    
    // MARK: - Table view delegate
    
    override func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
        self.performSegue(withIdentifier: self.items[indexPath.row].segue, sender: indexPath)
    }    
}
